chore(docs): Update proving backend related docs (#5601)
noir-lang/noir: docs/docs/getting_started/hello_noir/index.md

---

Now that we have installed Nargo and a proving backend, it is time to make our first hello world program!

### 1. Create a new project directory

Noir code can live anywhere on your computer. Let us create a _projects_ folder in the home
directory to house our first Noir program.

Create the directory and change directory into it by running:

```sh
mkdir ~/projects
cd ~/projects
```

## Nargo

Nargo provides the ability to initiate and execute Noir projects. Read the [Nargo installation](../installation/index.md) section to learn more about Nargo and how to install it.

### 2. Create a new Noir project

Now that we are in the projects directory, create a new Nargo project by running:

```sh
nargo new hello_world
```

`hello_world` can be any arbitrary project name, we are simply using `hello_world` for demonstration.

> In production, the common practice is to name the project folder as `circuits` for better identifiability when sitting alongside other folders in the codebase (e.g. `contracts`, `scripts`, `test`).

Reviewer: ```suggestion
In production, it is common practice to name the project folder, `circuits`, for clarity amongst other folders in the codebase (like: `contracts`, `scripts`, and `test`).
```

The witness corresponding to this execution will then be written to the file `./target/witness-name.gz`.

The command also automatically compiles your Noir program if it was not already / was edited, which you may notice the compiled artifacts being written to the file `./target/hello_world.json`.

## Proving Backend

Proving backends provide the ability to generate and verify proofs of executing Noir programs, following Noir's tooling that compiles and executes the programs. Read the [proving backend installation](../backend/index.md) section to learn more about proving backends and how to install them.

> Barretenberg is used as an example here to demonstrate how proving and verifying could be implemented and used. Read the [`bb` installation](../backend/index.md#example-installing-bb) section for how to install Barretenberg's CLI tool; refer to [`bb`'s documentations](https://github.com/AztecProtocol/barretenberg/blob/master/cpp/src/barretenberg/bb/readme.md) for full details about the tool.

Reviewer: ```suggestion
Barretenberg is used as an example here to demonstrate how proving and verifying could be implemented and used. Read the [`bb` installation](../backend/index.md#example-installing-bb) section for how to install Barretenberg's CLI tool; refer to [`bb`'s documentation](https://github.com/AztecProtocol/barretenberg/blob/master/cpp/src/barretenberg/bb/readme.md) for full details about the tool.
```

### 5. Prove an execution of the Noir program

Using Barretenberg as an example, prove the valid execution of your Noir program running:

```sh
bb prove -b ./target/hello_world.json -w ./target/witness-name.gz -o ./target/proof
```

The proof generated will then be written to the file `./target/proof`.

### 6. Verify the execution proof

Once a proof is generated, we can verify correct execution of our Noir program by verifying the proof file.

> Using Barretenberg as an example, generate a verification key for your Noir program running:

Reviewer: ```suggestion
Using Barretenberg as an example, compute/output the verification key for the Noir program by running:
```
As I understand it, the proving key would already be generated, and the verification key is calculated from it.
